src/Storage/Controllers/InstancesController.cs

```diff
@@ -141,51 +141,53 @@ public class InstancesController : ControllerBase
                 return BadRequest("Both InstanceOwner.PartyId and InstanceOwnerIdentifier cannot be present at the same time.");
             }
 
-            if (orgClaim != null)
+            switch ((orgClaim != null, userId != null))
```

Comment:
I'm not sure turning the if into a switch was necessary. I would revert this.

Reply:
As we talked about. Instead of if-else do separate tests for different cases and exit early.

```
if (orgclaim is null and userid is null {
   return BadRequest()
}

If (orgClaim is not null) {
   validate and handle required parameters for org
}

Perform party "normalization"

If (user is not null) {
   validate and handle required parameters for user
}
```